ENIGMAsystem/SHELL/Graphics_Systems/General/GSprimitives.cpp

```diff
@@ -69,6 +69,9 @@ unsigned draw_primitive_count(int kind, unsigned vertex_count) {
 }
 
 void draw_batch_flush(int kind) {
+  if (kind == pr_undefined)
+    kind = draw_get_batch_mode();
```

Comment:
Who put this here? No no no no no, this is not correct at **ALL**. Kind is not primitive kind, it's batch kind (one of `immediate/deferred/never`).